pkg/build/controller/factory/factory.go

```diff
@@ -228,7 +228,7 @@ func (f *typeBasedFactoryStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.
 	case buildapi.CustomBuildStrategyType:
 		pod, err = f.CustomBuildStrategy.CreateBuildPod(build)
 	default:
-		return nil, errors.New("No strategy defined for type")
+		return nil, fmt.Errorf("no supported build strategy defined for build %s/%s with type %s", build.Namespace, build.Name, build.Parameters.Strategy.Type)
```

Comment:
This was already discussed above and we settled with @ncdc on - "on supported build ..."